server.js

```diff
@@ -18,8 +19,114 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
 });
+////// Tuesday ////
+
+const { Schema } = mongoose;
+const HappyThoughtsSchema = new Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5, 
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0 
+
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date()
+  },
+}); 
+
+const HappyThoughts = mongoose.model("HappyThoughts", HappyThoughtsSchema);
+
+app.get("/thoughts", async (req, res) => {
+  try {
+    const thoughts = await HappyThoughts.find().sort({ createdAt: -1 }).limit(20);
+    res.status(200).json({
+      success: true,
+      response: thoughts,
+      message: "Thoughts retrieved successfully"
+    });
+  } catch(e) {
+    res.status(400).json({
+      success: false,
+      response: e,
+      message: "did not successfully"
+     });
+  }
+});
+
+
+app.post("/thoughts", async (req, res)=>{
+const {message} = req.body;
+try{ 
+      const newThoughts = await new HappyThoughts({message}).save();
+      res.status(201).json({
+        success: true,
+        response: newThoughts,
+         message: "created successfully"
+       });
+     } catch (e) {
+       res.status(400).json({
+        success: false,
+        response: e,
+        message: "error occured"
+      });
+    }
+ });
+
+ // POST thoughts/:thoughtId/like
+ app.post("/thoughts/:thoughtId/like", async (req, res) => {
+  try {
+    const likes = await HappyThoughts.findOneAndUpdate(
+      { "_id": req.params.thoughtId }, // Corrected parameter name
+      { $inc: { "hearts": 1 } },
+      { new: true }
+    );
+    res.status(201).json(likes);
+  } catch (err) {
+    res.status(400).json({ message: "Could not save your like", error: err.errors });
```

Comment:
What does the "error: err.errors" do? Have not used the same approach, would be interesting to know!